crates/agglayer-aggregator-notifier/src/certifier/mod.rs

```diff
@@ -323,11 +324,47 @@ where
             }
         };
 
+        let aggchain_vkey = match certificate.aggchain_data {
+            AggchainData::ECDSA { .. } => None,
+            AggchainData::Generic { .. } => {
+                if certificate.custom_chain_data.len() < 2 {
+                    return Err(CertificationError::Types {
+                        source: agglayer_types::Error::InvalidCustomChainDataLength {
+                            expected_at_least: 2,
+                            actual: certificate.custom_chain_data.len(),
+                        },
+                    });
+                }
+
+                let mut selector = [0u8; 2];
+                selector.copy_from_slice(&certificate.custom_chain_data[0..2]);
+
+                let aggchain_vkey_selector: u16 = u16::from_be_bytes(selector);
```

Comment:
Nit: limit the scope of mutable variables

```suggestion
                let aggchain_vkey_selector = {
                    let mut selector = [0u8; 2];
                    selector.copy_from_slice(&certificate.custom_chain_data[0..2]);
                    u16::from_be_bytes(selector)
                };
```